File: LeetCodeTasks/3012_minimize-length-of-array-using-operations/solutions/python3/solution.py
from typing import List

class Solution:
    def minimumArrayLength(self, nums: List[int]) -> int:
        while len(nums) > 1:
            nums.sort()
            if nums[-2] == 0:
                break
            nums.append(nums[-2] % nums[-1])
            nums.pop()
            nums.pop()
        return len(nums)

File: LeetCodeTasks/3012_minimize-length-of-array-using-operations/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minimumArrayLength(vector<int>& nums) {
        while(nums.size() > 1) {
            sort(nums.begin(), nums.end());
            int remainder = nums.back() % nums[nums.size() - 2];
            nums.pop_back();
            nums.pop_back();
            if(remainder != 0)
                nums.push_back(remainder);
        }
        return nums.size();
    }
};
